src/Core/Grid/Definition/Factory/CartGridDefinitionFactory.php

```diff
@@ -0,0 +1,305 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\HighlightedColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DateTimeColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\DateRangeType;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Creates grid definition for Carts grid
+ */
+final class CartGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    const GRID_ID = 'cart';
+
+    /**
+     * @var ConfigurationInterface
+     */
+    private $configuration;
+
+    /**
+     * @var string
+     */
+    private $contextDateFormat;
+
+    /**
+     * @var AccessibilityCheckerInterface
+     */
+    private $deleteCartAccessibilityChecker;
+
+    /**
+     * @param HookDispatcherInterface $dispatcher
+     * @param ConfigurationInterface $configuration
+     * @param AccessibilityCheckerInterface $deleteCartAccessibilityChecker
+     * @param string $contextDateFormat
+     */
+    public function __construct(
+        HookDispatcherInterface $dispatcher,
+        ConfigurationInterface $configuration,
+        AccessibilityCheckerInterface $deleteCartAccessibilityChecker,
+        $contextDateFormat
+    ) {
+        parent::__construct($dispatcher);
+
+        $this->configuration = $configuration;
+        $this->contextDateFormat = $contextDateFormat;
+        $this->deleteCartAccessibilityChecker = $deleteCartAccessibilityChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Shopping Carts', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add((new BulkActionColumn('carts_bulk'))
+                ->setOptions([
+                    'bulk_field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('id_cart'))
+                ->setName($this->trans('ID', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'id_cart',
+                ])
+            )
+            ->add((new DataColumn('status'))
+                ->setName($this->trans('Order ID', [], 'Admin.Orderscustomers.Feature'))
```

Comment:
This column can contain either an order_id if the cart has been finalized, but also `abandoned_cart` or `not_ordered` So I guess the column name should state this fact:
```php
->setName($this->trans('Order ID / Status', [], 'Admin.Orderscustomers.Feature'))
```

@LouiseBonnard what do you think?